File: Canopy_Sentinal_G_Final_dashboard-master/app/src/main/java/com/erc/canopysentinalg/model/DeforestationAlert.java
package com.erc.canopysentinalg.model;

public class DeforestationAlert {
    private String region;
    private double area;
    private String timestamp;
    private double latitude;
    private double longitude;
    private double ndviValue;

    public DeforestationAlert(String region, double area, String timestamp, double latitude, double longitude, double ndviValue) {
        this.region = region;
        this.area = area;
        this.timestamp = timestamp;
        this.latitude = latitude;
        this.longitude = longitude;
        this.ndviValue = ndviValue;
    }

    // Getters
    public String getRegion() { return region; }
    public double getArea() { return area; }
    public String getTimestamp() { return timestamp; }
    public double getLatitude() { return latitude; }
    public double getLongitude() { return longitude; }
    public double getNdviValue() { return ndviValue; }

    // Notification message format
    public String getNotificationMessage() {
        return String.format("🚨 Deforestation Detected in %s! Estimated loss: %.1f ha on %s", 
            region, area, timestamp);
    }
} 

File: Canopy_Sentinal_G_Final_dashboard-master/app/src/main/java/com/erc/canopysentinalg/data/model/DeforestationAlert.kt
package com.erc.canopysentinalg.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class DeforestationAlert(
    val id: String,
    val region: String,
    val country: String? = null,
    val area: Double, // in hectares
    val timestamp: Long,
    val latitude: Double,
    val longitude: Double,
    val ndviValue: Double,
    val severity: AlertSeverity = AlertSeverity.MEDIUM,
    val imageUrl: String? = null
) : Parcelable {
    enum class AlertSeverity {
        LOW, MEDIUM, HIGH, CRITICAL
    }
    
    val notificationMessage: String
        get() = "🚨 Deforestation Detected in $region! Estimated loss: ${String.format("%.1f", area)} ha"
}
